### PR TITLE
Add test 'Filter sort and pagination order return status'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | no
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #{issue number here}.
| How to test?  | TEST_PATH=TEST_PATH="functional/BO/13_shopParameters/02_orderSettings/02_statuses/returnStatuses/02_filterSortAndPagination" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21779)

### CODE REVIEW

File: tests/UI/campaigns/functional/BO/13_shopParameters/02_orderSettings/02_statuses/returnStatuses/02_filterSortAndPagination.js [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+
+// Import data
+const {ReturnStatuses} = require('@data/demo/orderReturnStatuses');
+const OrderReturnStatusFaker = require('@data/faker/orderReturnStatus');
+
+// Import test context
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_shopParameters_orderSettings_statuses_'
+  + 'returnStatuses_filterSortAndPaginationOrderStatus';
```
Comment (SimonGrn) [inline tests/UI/campaigns/functional/BO/13_shopParameters/02_orderSettings/02_statuses/returnStatuses/02_filterSortAndPagination.js]: I need to check the database to see if I allowed enough characters... :upside_down_face:

**Review by SimonGrn — DISMISSED** [review]

File: tests/UI/campaigns/data/demo/orderReturnStatuses.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+module.exports = {
+  ReturnStatuses: {
+    waitingForConfirmation: {
+      id: 1,
+      name: 'Waiting for confirmation',
+    },
+    waitingForPackaging: {
```
Comment (Progi1984) [inline tests/UI/campaigns/data/demo/orderReturnStatuses.js]:
```suggestion
    waitingForPackage: {
```

File: tests/UI/campaigns/functional/BO/13_shopParameters/02_orderSettings/02_statuses/returnStatuses/02_filterSortAndPagination.js (hunk 1 above)
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/13_shopParameters/02_orderSettings/02_statuses/returnStatuses/02_filterSortAndPagination.js]:
```suggestion
  + 'returnStatuses_filterSortAndPagination';
```

**Review by Progi1984 — COMMENTED** [review]
Need to change baseContext in tests/UI/campaigns/functional/BO/13_shopParameters/02_orderSettings/02_statuses/statuses/01_CRUDOrderStatus.js

File: tests/UI/campaigns/data/demo/orderReturnStatuses.js (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/data/demo/orderReturnStatuses.js]: Done thx :+1:

File: tests/UI/campaigns/functional/BO/13_shopParameters/02_orderSettings/02_statuses/returnStatuses/02_filterSortAndPagination.js [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+
+let browserContext;
+let page;
+let numberOfOrderReturnStatuses = 0;
+const tableName = 'order_return';
+
+/*
+Filter order return status by : Id, Name
+Sort order return status by : Id, Name
+Create 2 order return statuses
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/13_shopParameters/02_orderSettings/02_statuses/returnStatuses/02_filterSortAndPagination.js]:
```suggestion
Create 16 order return statuses
```

File: tests/UI/campaigns/functional/BO/13_shopParameters/02_orderSettings/02_statuses/returnStatuses/02_filterSortAndPagination.js [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+        await expect(pageTitle).to.contains(addOrderReturnStatusPage.pageTitleCreate);
+      });
+
+      it('should create order status and check result', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', `createOrderReturnStatus${index}`, baseContext);
+
+        await addOrderReturnStatusPage.setOrderReturnStatus(page, orderReturnStatusData);
+        /* Successful message is not visible, skipping it */
+        /* https://github.com/PrestaShop/PrestaShop/issues/21270 */
+        // await expect(textResult).to.contains(statusesPage.successfulCreationMessage);
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/13_shopParameters/02_orderSettings/02_statuses/returnStatuses/02_filterSortAndPagination.js:235]: Do we have a list of skipped steps or actions like this ?
Could be done in skipped test file. (to keep track and change it when it's fixed)

File: tests/UI/campaigns/functional/BO/13_shopParameters/02_orderSettings/02_statuses/returnStatuses/02_filterSortAndPagination.js (hunk 3 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/BO/13_shopParameters/02_orderSettings/02_statuses/returnStatuses/02_filterSortAndPagination.js]: Done thx :+1:

**Review by Progi1984 — COMMENTED** [review]
Need to change baseContext in tests/UI/campaigns/functional/BO/13_shopParameters/02_orderSettings/02_statuses/statuses/01_CRUDOrderStatus.js

**Comment by SimonGrn** [comment]
> Need to change baseContext in tests/UI/campaigns/functional/BO/13_shopParameters/02_orderSettings/02_statuses/statuses/01_CRUDOrderStatus.js

Is there a reason ?

**Comment by Progi1984** [comment]
The file has been renamed.

**Comment by nesrineabdmouleh** [comment]
> The file has been renamed.

I think that the name is correct, what's the problem?

**Review by SimonGrn — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**